tests/test_net_connect.js

```diff
@@ -19,8 +19,7 @@ describe('`disableNetConnect()`', () => {
 
     await assertRejects(
       got('https://other.example.test/'),
-      Error,
-      'Nock: Disallowed net connect for "other.example.test:443/"'
```

Comment:
This is the bug I mentioned in the PR description - this last parameter is used for displaying a message when the assertion fails, not to check the error itself.